FinSearchUnified/Tests/Subscriber/FrontendTest.php

```diff
@@ -157,54 +156,86 @@ public function testSessionValuesAfterListingCount($sSearch, $sCategory, $sContr
             ->setParams(['sSearch' => $sSearch, 'sCategory' => $sCategory]);
 
         // Create mocked Subject to be passed in mocked args
-        $subject = $this->getMockBuilder('\Enlight_Controller_Action')
+        $subject = $this->getMockBuilder(Enlight_Controller_Action::class)
             ->setMethods(['Request'])
             ->disableOriginalConstructor()
             ->getMock();
         $subject->method('Request')
             ->willReturn($request);
 
         // Create mocked args for getting Subject and Request
-        $args = $this->getMockBuilder('\Enlight_Controller_ActionEventArgs')
+        $args = $this->getMockBuilder(Enlight_Event_EventArgs::class)
             ->setMethods(['getSubject', 'getRequest'])
             ->getMock();
         $args->method('getSubject')->willReturn($subject);
         $args->method('getRequest')->willReturn($request);
 
-        $sessionArray = [
-            ['isSearchPage', isset($sSearch)],
-            ['isCategoryPage', isset($sCategory)]
-        ];
-
-        // Create Mock object for Shopware Session
-        $session = $this->getMockBuilder('\Enlight_Components_Session_Namespace')
-            ->setMethods(['offsetGet'])
-            ->getMock();
-        $session->expects($this->atLeastOnce())
-            ->method('offsetGet')
-            ->willReturnMap($sessionArray);
-
-        // Assign mocked session variable to application container
-        Shopware()->Container()->set('session', $session);
-
-        $isCategoryPage = Shopware()->Session()->offsetGet('isCategoryPage');
-        $isSearchPage = Shopware()->Session()->offsetGet('isSearchPage');
-
-        /** @var Plugin $plugin */
-        $plugin = Shopware()->Container()->get('kernel')->getPlugins()['FinSearchUnified'];
-        $frontend = new \FinSearchUnified\Subscriber\Frontend($plugin->getPath(), new Enlight_Template_Manager());
+        Shopware()->Session()->isCategoryPage = $isCategory;
+        Shopware()->Session()->isSearchPage = $isSearch;
 
+        $frontend = Shopware()->Container()->get('fin_search_unified.subscriber.frontend');
         $frontend->onFrontendPreDispatch($args);
 
+        // Check session values after FrontendPreDispatch Call
+        $isCategoryPage = Shopware()->Session()->isCategoryPage;
+        $isSearchPage = Shopware()->Session()->isSearchPage;
+
         $this->assertEquals(
+            $isCategory,
```

Comment:
Check the argument list of method `assertEquals` and make sure that `$isCategory` and `$isCategoryPage` are in the correct order.